src/lib/DutchDecayLib.sol

```diff
@@ -28,7 +28,9 @@ library DutchDecayLib {
         view
         returns (uint256 decayedAmount)
     {
-        if (decayEndTime <= decayStartTime) {
+        if (startAmount == endAmount) {
```

Comment:
a little strange that the decay times are not validated if startAmount==endAmount

Reply:
wdyt about putting this check after the time based ones?

Reply:
part of the point of this issue is that the time-based ones are irrelevant if startamount == endamount, which is always true for one of inputs or outputs. timings will be validated for whichever one is decaying